### ISSUE TITLE
Transpile references to BASH_SOURCE variable

### ISSUE THREAD

**Issue by Brassrat** [nfischer/shelljs-transpiler#36]
Seems that $0 is like 'sh'?
shouldn't it be the path to the '.js' file?
preferably some absolute version of the path to the js file?

**Comment by nfischer** [comment]
I think it's the path to the script, as specified on the CLI:

```sh
$ cat print0.sh
#!/bin/bash
echo $0
$ bash print0.sh # notice the output is exactly what I type here
print0.sh
$ bash ./print0.sh # again, it prints whatever I typed
./print0.sh
```

So, it's not guaranteed to be an absolute path.

**Comment by Brassrat** [comment]
yes $0 is not guaranteed to be an absolute path in BASH - for that one uses either BASH_SOURCE or something like (cd -P $0)

**Comment by nfischer** [comment]
FWIW I think we implement this correctly. We translate `$0` to `process.argv[1]`, which is the path to the script file. Note that `process.argv[0]` is the path to NodeJS itself.

Closing this because I think it's WAI, but let me know if I'm mistaken.

**Comment by Brassrat** [comment]
i suggested implementing $0 as the absolute path to obviate the need to support BASH_SOURCE[0].
Admittedly, node provides __dirname.

**Comment by nfischer** [comment]
I think `$0` is right as-is, but I might accept a PR for `BASH_SOURCE`, if you know how to implement all the entries and can point to docs.

Renaming this issue to reflect that.
